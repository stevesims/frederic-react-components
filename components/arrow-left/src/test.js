import React from 'react';
import { shallow } from 'enzyme';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

import Component from '.';

expect.extend(createMatchers(emotion));

describe('ArrowLeft', () => {
  let wrapper;
  
  it('renders without crashing', () => {
    wrapper = shallow(<Component />);
  });

  it('renders with a title', () => {
    wrapper = shallow(<Component>Example title</Component>);
    expect(wrapper.find('title').text()).toBe('Example title');
  });

  it('renders with a colour fill', () => {
    wrapper = shallow(<Component fill="red" />);
    expect(wrapper.find('svg[fill="red"]').length).toBe(1);
  });

  it('supports setting of no color', () => {
    expect(wrapper).not.toHaveStyleRule('color', 'purple');
  });

  it('supports setting of color', () => {
    wrapper = shallow(<Component color="purple" />);
    expect(wrapper).toHaveStyleRule('color', 'purple');
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
