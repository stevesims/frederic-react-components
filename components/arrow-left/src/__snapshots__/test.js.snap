// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowLeft matches snapshot 1`] = `
<Wrapper
  color="purple"
>
  <svg
    fill="#005ea5"
    height="20px"
    viewBox="0 0 512 512"
    width="20px"
  >
    <title />
    <path
      d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm28.9-143.6L209.4 288H392c13.3 0 24-10.7 24-24v-16c0-13.3-10.7-24-24-24H209.4l75.5-72.4c9.7-9.3 9.9-24.8.4-34.3l-11-10.9c-9.4-9.4-24.6-9.4-33.9 0L107.7 239c-9.4 9.4-9.4 24.6 0 33.9l132.7 132.7c9.4 9.4 24.6 9.4 33.9 0l11-10.9c9.5-9.5 9.3-25-.4-34.3z"
    />
  </svg>
</Wrapper>
`;
